6_nodejs/week3/app.js

```diff
@@ -0,0 +1,65 @@
+import knex from "knex";
+const knexInstance = knex({
+  client: "mysql2",
+  connection: {
+    host: process.env.DB_HOST || "127.0.0.1",
+    port: process.env.DB_PORT || 3307,
+    user: process.env.DB_USER || "root",
+    password: process.env.DB_PASSWORD || "pass",
+    database: process.env.DB_NAME || "hyf_node_week3_warmup",
+    multipleStatements: true,
+  },
+});
+
+import express from "express";
+const app = express();
+const port = process.env.PORT || 3003;
+
+app.use(express.json());
+
+const apiRouter = express.Router();
+app.use("/api", apiRouter);
+
+const contactsAPIRouter = express.Router();
+apiRouter.use("/contacts", contactsAPIRouter);
+
+/* SQL Injection
+SQL injection happens when the user, instead of inputting the valid data, 
+inputs a SQL statement that ultimately gets executed on the database. 
+The best defense against these kinds of attacks are the framework-supported, 
+SQL-prepared statements or using named parameters. 
+*/
+
+// SQL injection cheat sheet - https://portswigger.net/web-security/sql-injection/cheat-sheet
+
+contactsAPIRouter.get("/", async (req, res) => {
+  let query = knexInstance.select("*").from("contacts");
+
+  if ("sort" in req.query) {
+    const orderBy = req.query.sort.toString();
+    // preventing SQL injection using parameterized queries instead of string concatenation
+    const preparedStatementQueries = [
```

Comment:
Clever solution! 🧠 However, we're missing some functionality: you are not able to specify whether it is ascending or descending! 😮 Can you think of a way to support this without introducing a security issue?